src/vos/vos_common.c

```diff
@@ -325,7 +325,21 @@ vos_tls_fini(void *data)
 {
 	struct vos_tls *tls = data;
 
-	D_ASSERT(d_list_empty(&tls->vtl_gc_pools));
+	/* All GC callers should have exited, but they can still leave
+	 * uncleaned pools behind. It is OK to free these pool handles with
+	 * leftover, because GC can clean up leftover when it starts again.
+	 */
+	D_ASSERTF(tls->vtl_gc_running == 0, "GC running = %d\n",
+		  tls->vtl_gc_running);
```

Comment:
Is this 'vtl_gc_running' for checking if GC ULTs exited? I think wait_all_exited() on server shutdown has ensured all ULTs exited before calling TLS fini, so it looks not necessary to me.

Reply:
just in case :-)